pkg/systemd/service-details.jsx

```diff
@@ -0,0 +1,445 @@
+                        <a role="menuitem" onClick={() => this.props.actionCallback("RestartUnit")}>{ _("Restart") }</a>
+                    </li>,
+                    <li role="presentation" key="reload">
+                        <a role="menuitem" onClick={() => this.props.actionCallback("ReloadUnit")}>{ _("Reload") }</a>
+                    </li>
+                );
+
+            if (actions.length > 0)
+                actions.push(
+                    <li className="divider" role="presentation" key="divider2" />
+                );
+
+            actions.push(
+                <li role="presentation" key="mask">
+                    <a role="menuitem" onClick={() => this.setState({ dialogMaskedOpened: true }) }>{ _("Disallow running (mask)") }</a>
+                </li>
+            );
+        }
+
+        return (
+            <React.Fragment>
+                { this.state.dialogMaskedOpened &&
+                    <ServiceConfirmDialog title={ _("Mask Service") }
+                                          message={ _("Hey, be careful...")}
+                                          close={() => this.setState({ dialogMaskedOpened: false }) }
+                                          confirmText={ _("Mask Service") }
+                                          confirmAction={() => { this.props.fileActionCallback("MaskUnitFiles", false); this.setState({ dialogMaskedOpened: false }) }} />
+                }
+                <span className="dropdown">
+                    <a data-toggle="dropdown">
+                        <span className="fa fa-ellipsis-v bigger-clickable-area" />
+                    </a>
+                    <ul role="menu" className="dropdown-menu">
+                        {actions}
+                    </ul>
+                </span>
+            </React.Fragment>
+        );
+    }
+}
+ServiceActions.propTypes = {
+    masked: PropTypes.bool.isRequired,
+    active: PropTypes.bool.isRequired,
+    enabled: PropTypes.bool.isRequired,
+    isStatic: PropTypes.bool.isRequired,
+    actionCallback: PropTypes.func.isRequired,
+    fileActionCallback: PropTypes.func.isRequired,
+};
+
+/*
+ * React template for a service details
+ * Shows current status and informations about the service.
+ * Enables user to control this unit like starting, enabling, etc. the service.
+ * Required props:
+ *  -  unit
+ *      Unit as returned from systemd dbus API
+ *  -  permitted
+ *      True if user can control this unit
+ *  -  actionCallback
+ *      Method for calling unit methods like `UnitStart`
+ *  -  fileActionCallback
+ *      Method for calling unit file methods like `EnableUnitFiles`
+ * Optional props:
+ *  -  originTemplate
+ *      Template name, from which this unit has been initialized
+ */
+export class ServiceDetails extends React.Component {
+    constructor(props) {
+        super(props);
+        this.onOnOffSwitch = this.onOnOffSwitch.bind(this);
+    }
+
+    onOnOffSwitch() {
```

Comment:
With this, we're intentionally moving to a control that doesn't express it's full intent as much as the old dropdown did, for good and bad. I think it could help with a tooltip of what's it's doing. Like _Start and Enable_ or something like that.

Reply:
Done, there is tooltip.